### PR TITLE
Feature: APP-800 Add/Remove execution widget

### PR DESCRIPTION
**Pull request by sepehr2github**
## Description

- Add/Remove execution widget added to the proposal page

Task: [APP-800](https://aragonassociation.atlassian.net/browse/APP-800)

## Type of change

- [x] New feature (non-breaking change which adds functionality)

## Checklist:

- [x] I have selected the correct base branch.
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [x] Any dependent changes have been merged and published in downstream modules.
- [x] I ran all tests with success and extended them if necessary.
- [ ] I have updated the ``CHANGELOG.md`` file in the root folder.
- [x] I have tested my code on the test network.

### CODE REVIEW

**Review by RakeshUP — APPROVED** [review]

File: packages/web-app/src/components/executionWidget/withdrawCard.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import React from 'react';
```
Comment (WissenIstNacht) [inline packages/web-app/src/components/executionWidget/withdrawCard.tsx]: This is just code that was moved, right? It was under `reviewExecution/actions/withdrawCard.tsx` before?

If so, please use the `git move` command to rename/replace files.

File: packages/web-app/src/components/executionWidget/index.tsx [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+export const ReviewExecution: React.FC = () => {
+  const {t} = useTranslation();
+  const {getValues} = useFormContext();
+  const {actions} = getValues();
+  const {actions: actionTypes} = useActionsContext();
+
+  const {data: daoId} = useDaoParam();
+  const {data: dao} = useDaoMetadata(daoId);
+
+  const RenderActions = useCallback(
```
Comment (WissenIstNacht) [inline packages/web-app/src/components/executionWidget/index.tsx]: This seems to be the old file. Please make sure you don't introduce errors when moving/renaming things (again, using git helps here).

**Review by WissenIstNacht — CHANGES_REQUESTED** [review]
Hey, there seem to be issues with some versions of some files in this PR. Also, the diff shows things that should already be in develop...? Could you please update it such that it matches the current dev branch?

File: packages/web-app/src/components/executionWidget/actions/addAddressCard.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import {ActionAddAddress} from 'utils/types';
+import {AccordionMethod} from 'components/accordionMethod';
+import {ListItemAddress} from '@aragon/ui-components';
+import AccordionSummary from 'containers/actionBuilder/addAddresses/accordionSummary';
+
+export const AddAddressCard: React.FC<{
+  action: ActionAddAddress;
+}> = ({action}) => {
+  const {t} = useTranslation();
+  const filteredMemberWallets = action.inputs.memberWallets.filter(
```
Comment (WissenIstNacht) [inline packages/web-app/src/components/executionWidget/actions/addAddressCard.tsx:14]: This is to make sure that "empty addresses" are not counted?

File: packages/web-app/src/components/executionWidget/actions/addAddressCard.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    <AccordionMethod
+      type="execution-widget"
+      methodName={t('labels.addWallets')}
+      smartContractName={t('labels.aragonCore')}
+      verified
+      methodDescription={t('labels.addWalletsDescription')}
+    >
+      <Container>
+        {filteredMemberWallets.map(({address}, index: number) => (
+          <ListItemAddress label={address} src={address} key={index} />
```
Comment (WissenIstNacht) [inline packages/web-app/src/components/executionWidget/actions/addAddressCard.tsx]: The `address` would be better as a key.

File: packages/web-app/src/containers/actionBuilder/addAddresses/accordionSummary.tsx [hunk 5]
```diff
@@ -1,25 +1,65 @@
 import React from 'react';
 import styled from 'styled-components';
 import {useTranslation} from 'react-i18next';
+import {AccordionMethodType, AccordionType} from 'components/accordionMethod';
+import {IconLinkExternal, Link} from '@aragon/ui-components';
 
-const AccordionSummary: React.FC<{total: number}> = ({total}) => {
+type AccordionSummaryPropsType = {
+  type?: AccordionMethodType['type'];
```
Comment (WissenIstNacht) [inline packages/web-app/src/containers/actionBuilder/addAddresses/accordionSummary.tsx:8]: Technically I think you now export a type that does just that :p

File: packages/web-app/src/components/executionWidget/actions/removeAddressCard.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  return (
+    <AccordionMethod
+      type="execution-widget"
+      methodName={t('labels.removeWallets')}
+      smartContractName={t('labels.aragonCore')}
+      verified
+      methodDescription={t('labels.removeWalletsDescription')}
+    >
+      <Container>
+        {action.inputs.memberWallets.map(({address}, index: number) => (
```
Comment (WissenIstNacht) [inline packages/web-app/src/components/executionWidget/actions/removeAddressCard.tsx]: same

**Review by WissenIstNacht — APPROVED** [review]
Aaah, much better, thanks for updating the branch :D

Would be nice if you could swap out the keys on the mappings. Otherwise, approved!

File: packages/web-app/src/components/executionWidget/actions/addAddressCard.tsx (hunk 3 above)
Comment (sepehr2github) [reply packages/web-app/src/components/executionWidget/actions/addAddressCard.tsx:14]: Yup!

File: packages/web-app/src/components/executionWidget/actions/addAddressCard.tsx (hunk 4 above)
Comment (sepehr2github) [reply packages/web-app/src/components/executionWidget/actions/addAddressCard.tsx]: Ah true Thanks

**Pull request merged**